### PR TITLE
Bug/1716 bug in v2 entities without options key values

### PR DESCRIPTION
**Pull request by kzangeli**
Two "bugs" fixed;
  - Issue #1716 
  - Issue #1713 

Also, a minor error in harnessFunctions.sh fixed (about accessing a file before knowing it actually exists).

### CODE REVIEW

File: src/lib/ngsi/Metadata.cpp [hunk 1]
```diff
@@ -372,6 +372,10 @@ std::string Metadata::toJson(bool isLastElement)
   {
     out += JSON_VALUE_BOOL("value", boolValue);
   }
+  else if (valueType == orion::ValueTypeNone)
+  {
+    out += JSON_STR("value") + ":" + "null";
```
Comment (crbrox) [inline src/lib/ngsi/Metadata.cpp]: +":null" could be an improvement

File: test/functionalTest/cases/1713_null_value_for_metadatas/null_value_for_metadatas.test [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        "type": null,
+        "value": "34"
+    },
+    "type": "room"
+}
+
+
+--TEARDOWN--
+brokerStop CB
+#dbDrop CB
```
Comment (fgalan) [inline test/functionalTest/cases/1713_null_value_for_metadatas/null_value_for_metadatas.test]: Remove #

**Comment by fgalan** [comment]
LGTM (passing the ball to @crbrox for final LGTM + merge + delete branch)

File: src/lib/ngsi/Metadata.cpp (hunk 1 above)
Comment (kzangeli) [inline src/lib/ngsi/Metadata.cpp]: Fixed in 5d8ecea.

A little more ugly, but more efficient :-)

File: test/functionalTest/cases/1713_null_value_for_metadatas/null_value_for_metadatas.test (hunk 2 above)
Comment (kzangeli) [inline test/functionalTest/cases/1713_null_value_for_metadatas/null_value_for_metadatas.test]: Always these ...
Fixed in 5d8ecea

**Comment by crbrox** [comment]
LGTM

**Pull request merged**